xarray/tests/test_dataset.py

```diff
@@ -6197,16 +6169,16 @@ def test_dir_non_string(data_set):
     dir(x2)
 
 
-def test_dir_unicode(data_set):
-    data_set["unicode"] = "uni"
-    result = dir(data_set)
+def test_dir_unicode(ds):
+    ds["unicode"] = "uni"
+    result = dir(ds)
     assert "unicode" in result
 
 
 @pytest.fixture(params=[1])
-def ds(request):
+def ds(request, backend):
```

Comment:
This fixture is less elegant than I'd hoped; it now has three variations.

I think we could attempt to define one (maybe a couple) canonical datasets, and aim to use those any new tests that aren't testing something specific (e.g. some will need all missing data, but most tests have similar requirements). 

Potentially that's just the result of the `create_test_data` function — is that our canonical dataset?